api/src/test/java/org/apache/iceberg/io/TestCloseableIterable.java

```diff
@@ -201,6 +201,14 @@ public CloseableIterator<Integer> iterator() {
     Assertions.assertThat(counter.get()).isEqualTo(items.size());
   }
 
+  @Test
+  public void concatIterablesWithIterator() throws IOException {
```

Comment:
nit: this method name can be more descriptive, sth like `testCloseWithEmptyIterables`

Reply:
I specifically avoid naming it like this, because the important piece where this NPE is happening is the usage of `.iterator()` before doing a `.close()`